pkg/openapi/crdSync.go

```diff
@@ -107,17 +107,12 @@ func (c *crdSync) sync() {
 }
 
 func (o *Controller) deleteCRDFromPreviousSync() {
-	for k := range o.kindToDefinitionName {
-		delete(o.kindToDefinitionName, k)
+	for _, crd := range o.crdList {
+		delete(o.kindToDefinitionName, crd)
+		delete(o.definitions, crd)
 	}
 
-	o.kindToDefinitionName = make(map[string]string, 0)
-
-	for k := range o.definitions {
-		delete(o.definitions, k)
-	}
-
-	o.definitions = make(map[string]*openapi_v2.Schema, 0)
+	o.crdList = make([]string, 0)
```

Comment:
Any reason why we need to maintain a CRD list? I think I had removed it as it was not being used....did I miss something?

Reply:
So there are 2 types of schemas store in the `kindToDefinitionName` map, 1.) k8s native objects; 2.) CRDs. For each reconciliation of CRDController, it drops the cache for CRD definitions and syncs the latest from the cluster, while leaving the native resources definition unchanged. This CRD list is used to delete the entry from the cache.

With the recent changes, we cleaned up all definitions, that's why we are seeing "couldn't parse schema" error for Namespace, ConfigMap, etc.